src/components/show-products/index.js

```diff
@@ -43,19 +67,34 @@ export const ShowProducts = () => {
 
   return (
     <ContentContainer>
+      <form className="filter-form">
+        <label htmlFor="filter">
+          Filter items
+          <input
+            className="input-item"
+            type="search"
+            id="filter"
+            name="q"
+            aria-label="Search through list content"
+            placeholder="Start typing here..."
+            value={searchTerm}
+            onChange={(event) => setSearchTerm(event.target.value)}
+          />
+        </label>
+      </form>
       <ul>
-        {products?.map(({ id, productName, lastPurchaseDate, timeFrame }) => (
+        {list.map(({ id, productName, lastPurchaseDate, timeFrame }) => (
```

Comment:
Using [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) would help for weird cases where _list_ might be empty.
```suggestion
        {list?.map(({ id, productName, lastPurchaseDate, timeFrame }) => (
```